tiledb/sm/query/readers/dense_reader.cc

```diff
@@ -432,7 +432,7 @@ Status DenseReader::dense_read() {
     // prevent using too much memory when the budget is small and doesn't allow
     // to process more than one batch at a time.
     if (wait_compute_task_before_read && compute_task.valid()) {
-      throw_if_not_ok(resources_.compute_tp().wait(compute_task));
+      throw_if_not_ok(resources_.compute_tp().wait(&compute_task));
```

Comment:
Is the plan to replace all calls to `tp.wait(task)` with `task.wait()`?

Reply:
yes, I can do it already if you think it should be in this PR, or in the following work as I was planning.

Reply:
Hmm, on the one hand I prefer the existing APIs because the action primarily happens on the thread pool instead of the task, but on the other hand there is already a `wait()` method on `std::future` that we must be careful not to call because of deadlocks. 🤔

Reply:
Right, and also having to pass around a reference to the threadpool object everywhere is quite cumbersome.